homeassistant/components/fan/wemo.py

```diff
@@ -111,29 +117,44 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     def service_handle(service):
         """Handle the WeMo humidifier services."""
         entity_ids = service.data.get(ATTR_ENTITY_ID)
-        target_humidity = service.data.get(ATTR_TARGET_HUMIDITY)
+
+        humidifiers = []
 
         if entity_ids:
-            humidifiers = [device for device in hass.data[DATA_KEY].values() if
-                           device.entity_id in entity_ids]
+            humidifiers.extend([device for device in
+                                hass.data[DATA_KEY].values() if
+                                device.entity_id in entity_ids])
+
+            if service.service == SERVICE_SET_HUMIDITY:
+                target_humidity = service.data.get(ATTR_TARGET_HUMIDITY)
+
+                for humidifier in humidifiers:
+                    humidifier.set_humidity(target_humidity)
+            elif service.service == SERVICE_RESET_FILTER_LIFE:
+                for humidifier in humidifiers:
+                    humidifier.reset_filter_life()
         else:
-            humidifiers = hass.data[DATA_KEY].values()
-
-        for humidifier in humidifiers:
-            humidifier.set_humidity(target_humidity)
+            _LOGGER.error("Unable to execute service %s %s",
+                          service.service,
+                          "- entity_ids is a required field.")
```

Comment:
Why put this in a string substitution? Indentation?

```python
_LOGGER.error("Unable to execute service %s "
              "- entity_ids is a required field.",
              service.service)
```

Reply:
Ah, yep. I'll fix that.